### PR TITLE
NodeDefDetails: refactored BasicProps

### PR DESCRIPTION
**Pull request by SteRiccio**

### CODE REVIEW

File: webapp/components/survey/NodeDefDetails/store/actions/useValidateNodeDef.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  const state = getState()
+  const survey = SurveyState.getSurvey(state)
+
+  // Update survey with updated node def
+  const surveyUpdated = R.pipe(
+    // Associate updated node def
+    Survey.assocNodeDef(nodeDef),
+    // Build and associate dependency graph
+    Survey.buildAndAssocDependencyGraph
+  )(survey)
```
Comment (ramirobg94) [inline webapp/components/survey/NodeDefDetails/store/actions/useValidateNodeDef.js:25]: how about to extract this to local function like:

```
const assocNodeDefAndDependencyGraph = ({ nodeDef }) => (survey) => ...
```

in that way, the code could be more readable

File: webapp/components/survey/NodeDefDetails/store/hooks/useBasicProps.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  const survey = useSurvey()
+  const surveyCycleKey = useSurveyCycleKey()
+
+  const nodeDef = NodeDefState.getNodeDef(nodeDefState)
+  const validation = NodeDefState.getValidation(nodeDefState)
+
+  const isEntityAndNotRoot = NodeDef.isEntity(nodeDef) && !NodeDef.isRoot(nodeDef)
+  const displayAsEnabled = isEntityAndNotRoot
+  const displayInEnabled = isEntityAndNotRoot
+  const displayAsFormDisabled = false
```
Comment (ramirobg94) [inline webapp/components/survey/NodeDefDetails/store/hooks/useBasicProps.js:23]: maybe this variable and its use is not needed

File: webapp/components/survey/NodeDefDetails/store/hooks/useBasicProps.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  const displayAsTableDisabled = Survey.hasNodeDefChildrenEntities(nodeDef)(survey) || NodeDef.isSingle(nodeDef)
+
+  const displayInParentPageDisabled = NodeDefLayout.isRenderForm(surveyCycleKey)(nodeDef)
+  const displayInOwnPageDisabled = false
```
Comment (ramirobg94) [inline webapp/components/survey/NodeDefDetails/store/hooks/useBasicProps.js:27]: maybe this variable and its use is not needed

File: webapp/components/survey/NodeDefDetails/store/actions/useValidateNodeDef.js (hunk 1 above)
Comment (SteRiccio) [reply webapp/components/survey/NodeDefDetails/store/actions/useValidateNodeDef.js:25]: uhm, you are right, I'll extract that part in a function

File: webapp/components/survey/NodeDefDetails/store/hooks/useBasicProps.js (hunk 2 above)
Comment (SteRiccio) [reply webapp/components/survey/NodeDefDetails/store/hooks/useBasicProps.js:23]: are you talking about `displayAsFormDisabled`? I was asking to my self the same thing... :)
I'll remove it.

File: webapp/components/survey/NodeDefDetails/store/hooks/useBasicProps.js (hunk 3 above)
Comment (SteRiccio) [reply webapp/components/survey/NodeDefDetails/store/hooks/useBasicProps.js:27]: uhm, yes, same story as above.

File: webapp/components/survey/NodeDefDetails/store/actions/useValidateNodeDef.js (hunk 1 above)
Comment (minotogna) [reply webapp/components/survey/NodeDefDetails/store/actions/useValidateNodeDef.js:25]: or even better pass a parameter:

```const assocNodeDef = ({ nodeDef, updateDependencyGraph: false }) => (survey) => ...```

File: webapp/components/survey/NodeDefDetails/BasicProps.js [hunk 4]
```diff
@@ -206,14 +188,11 @@ const BasicProps = (props) => {
...
-  editingNodeDefFromDesigner: PropTypes.bool.isRequired,
-
-  setNodeDefParentUuid: PropTypes.func.isRequired,
-  setNodeDefProp: PropTypes.func.isRequired,
-  setNodeDefLayoutProp: PropTypes.func.isRequired,
+  nodeDefState: PropTypes.object.isRequired,
+  setNodeDefState: PropTypes.func.isRequired,
+  keyEditDisabled: PropTypes.bool.isRequired,
+  multipleEditDisabled: PropTypes.bool.isRequired,
+  editingFromDesigner: PropTypes.bool.isRequired,
```
Comment (minotogna) [inline webapp/components/survey/NodeDefDetails/BasicProps.js:195]: are these props used only here? if yes move them in its own state

Comment (SteRiccio) [reply webapp/components/survey/NodeDefDetails/BasicProps.js:191]: right

File: webapp/components/survey/NodeDefDetails/store/actions/useValidateNodeDef.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  const surveyUpdated = Survey.assocNodeDef({ nodeDef, updateDependencyGraph: true })(survey)
+
+  // Validate node def
+  const nodeDefValidation = await SurveyValidator.validateNodeDef(surveyUpdated, nodeDef)
+
+  // Update local state
+  setNodeDefState(NodeDefState.assocNodeDefAndValidation(nodeDef, nodeDefValidation)(nodeDefState))
+
+  // Dispatch update action
+  dispatch({ type: NodeDefsActions.nodeDefPropsUpdate, nodeDef })
```
Comment (minotogna) [inline webapp/components/survey/NodeDefDetails/store/actions/useValidateNodeDef.js:25]: better not to export keys here. but rather an action e.g,
```dispatch(NodeDefsActions.updateNodeDef({nodeDef }))```

Comment (SteRiccio) [reply webapp/components/survey/NodeDefDetails/store/actions/useValidateNodeDef.js:25]: ok

File: webapp/components/survey/NodeDefDetails/store/actions/useValidateNodeDef.js (hunk 1 above)
Comment (SteRiccio) [reply webapp/components/survey/NodeDefDetails/store/actions/useValidateNodeDef.js]: done

**Pull request merged**